### PR TITLE
Add find address command

### PR DESCRIPTION
**Pull request by ongweijie7**
Needs to be implemented so that we may search for health conditions/medicine usage in future.

Needs to include 'a'/'n' to indicate address attribute or name attribute before the actual arguments. Eg. "find n Bernice" will return all persons that contain the string "bernice"(case insensitive) in their names.

It is done this way to differentiate the different attributes that we might search for in future. Can be edit in future to improve the syntax of commands.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -89,21 +89,20 @@ Format: `delete i/NRIC`
...
-### Filtering a person: `filter`
+### Locating person by name or address: `find`
 
-Filters people according to a particular attribute. Eg. Medicine usage or health conditions.
+Find people according to a particular attribute. Eg. name(n) or address(a)
+Format: `find ATTRIBUTE KEYWORD [MORE_KEYWORDS]`
 
-Format: `filter KEYWORD [MORE_KEYWORDS]`
 
+* The search is case-sensitive. e.g panadol will match pANAdol
```
Comment (Creationsv2) [inline docs/UserGuide.md:98]: Should this be "not case-sensitive"?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -89,21 +89,20 @@ Format: `delete i/NRIC`
...
+* The order of the keywords does matters. e.g. panadol will match medicine panadol
```
Comment (Creationsv2) [inline docs/UserGuide.md:99]: a bit confusing

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 3]
```diff
@@ -3,31 +3,50 @@
...
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String property = matcher.group("property").trim();
+        final String trimmedArgs = matcher.group("arguments").trim();
+
         if (trimmedArgs.isEmpty()) {
             throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
```
Comment (Creationsv2) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:37]: How come we change it do the HelpCommand's invalid format? Shouldnt we keep it as FindCommand's

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 4]
```diff
@@ -3,31 +3,50 @@
...
 
         String[] nameKeywords = trimmedArgs.split("\\s+");
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        if (property.equals("a")) {
+            return new FindCommand(new AddressContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        } else if (property.equals("n")) {
+            return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        } else {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
```
Comment (Creationsv2) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:47]: Same here as above

File: src/test/java/seedu/address/logic/parser/FindCommandParserTest.java [hunk 5]
```diff
@@ -17,18 +19,37 @@ public class FindCommandParserTest {
 
     @Test
     public void parse_emptyArg_throwsParseException() {
-        assertParseFailure(parser, "     ", String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, "     ", String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
     }
 
     @Test
-    public void parse_validArgs_returnsFindCommand() {
+    public void parse_validArgs_returnsFindNameCommand() {
```
Comment (Creationsv2) [inline src/test/java/seedu/address/logic/parser/FindCommandParserTest.java:26]: Should we keep this test and implement another findNameCommand test?

**Review by Creationsv2 — CHANGES_REQUESTED** [review]
Logic looks good, just some minor questions and tweaks

File: docs/UserGuide.md [hunk 6]
```diff
@@ -89,21 +89,20 @@ Format: `delete i/NRIC`
 
 Examples: `delete i/T0012345A` to delete patient with NRIC T0012345A from patient records system
 
-### Filtering a person: `filter`
+### Locating person by name or address: `find`
```
Comment (CedricPei) [inline docs/UserGuide.md:92]: Locating a person ?

File: docs/UserGuide.md (hunk 2 above)
Comment (CedricPei) [reply docs/UserGuide.md:99]: I get it here but also think a clearer interpretation compared with the original one is better.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java (hunk 3 above)
Comment (CedricPei) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java:37]: Agreed to change it to HelpCommand's format. Good way to keep consistency.

File: docs/UserGuide.md (hunk 6 above)
Comment (ongweijie7) [reply docs/UserGuide.md:92]: I referenced the original user guide of AB tree for the find command but will change it to find if you think that is better

File: src/test/java/seedu/address/logic/parser/FindCommandParserTest.java (hunk 5 above)
Comment (ongweijie7) [reply src/test/java/seedu/address/logic/parser/FindCommandParserTest.java:26]: Actually, the test was kept. What I did here was to split the testing into 2 different test functions and named them better to differentiate between the two find functionalities.
Screenshot 2023-03-13 at 3 21 41 PM

File: docs/UserGuide.md (hunk 2 above)
Comment (ongweijie7) [reply docs/UserGuide.md]: yeap ok I will make the necessary changes

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java (hunk 3 above)
Comment (ongweijie7) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Yeap makes sense I will make the necessary changes

**Review by Creationsv2 — APPROVED** [review]
LGTM.

**Pull request merged**